examples/lerax-square.cpp

```diff
@@ -53,75 +67,35 @@ struct Player {
     }
 };
 
-class Game {
+
+class Game: public GameLoop {
 private:
-    SDL_Window *window = nullptr;
-    SDL_Renderer *renderer = nullptr;
-    const char *title = "Square Moving";
-    bool running = true;
-    struct Player player;
+    using GameLoop::GameLoop;
+    struct Player player = {};
     struct KeyboardState keyboard = {};
-
-    void event(SDL_Event *e) {
-        while (SDL_PollEvent(e)) {
-            switch (e->type) {
-            case SDL_QUIT:
-                running = false;
-                break;
-            case SDL_KEYDOWN:
-                switch(e->key.keysym.sym) {
-                case SDLK_ESCAPE:
-                    running = false;
-                    break;
-                case SDLK_UP:
-                    keyboard.up = true;
-                    break;
-                case SDLK_DOWN:
-                    keyboard.down = true;
-                    break;
-                case SDLK_LEFT:
-                    keyboard.left = true;
-                    break;
-                case SDLK_RIGHT:
-                    keyboard.right = true;
-                    break;
-                case SDLK_w:
-                    keyboard.velocity_up = true;
-                    break;
-                case SDLK_s:
-                    keyboard.velocity_down = true;
-                    break;
-                }
-                break;
-            case SDL_KEYUP:
-                switch(e->key.keysym.sym) {
-                case SDLK_UP:
-                    keyboard.up = false;
-                    break;
-                case SDLK_DOWN:
-                    keyboard.down = false;
-                    break;
-                case SDLK_LEFT:
-                    keyboard.left = false;
-                    break;
-                case SDLK_RIGHT:
-                    keyboard.right = false;
-                    break;
-                }
-                break;
-            }
-        }
+    InputHandler input_handler = InputHandler(input_config);
+
+    void event() {
+        static SDL_Event e;
+        input_handler.process(e);
+        keyboard.up = input_handler.read("up");
+        keyboard.down = input_handler.read("down");
+        keyboard.left = input_handler.read("left");
+        keyboard.right = input_handler.read("right");
+        keyboard.velocity_up = input_handler.read("velocity_up");
+        keyboard.velocity_down = input_handler.read("velocity_down");
+        running = !(input_handler.read(SDL_QUIT) || input_handler.read("quit"));
     }
 
     void update(float dt) {
         if (keyboard.velocity_up) {
             player.velocity += 50;
-            keyboard.velocity_up = false;
+            input_handler.write("velocity_up", false);
```

Comment:
Yep, referring to the previous comment about this `InputHandler` class I can see why you have done it that way. Well, events are either read or written inside a buffer, be that buffer a cyclic one or not. This is quite good for now.